vm/devices/storage/disk_nvme/nvme_driver/src/queue_pair.rs

```diff
@@ -518,6 +636,29 @@ impl QueueHandler {
             }
         }
     }
+
+    /// Save queue data for servicing.
+    pub async fn save(&self) -> anyhow::Result<QueuePairSavedState> {
+        // The data is collected from both QueuePair and QueueHandler.
+        Ok(QueuePairSavedState {
+            sq_state: self.sq.save(),
+            cq_state: self.cq.save(),
+            pending_cmds: self.commands.save(),
+            cpu: 0,       // Will be updated by the caller.
```

Comment:
This is an error-prone technique. Can we just nest this structure to avoid this "caller has to remember to fill stuff in" behavior?

Reply:
Done